app/models/generic_object_definition.rb

```diff
@@ -0,0 +1,12 @@
+class GenericObjectDefinition < ApplicationRecord
+  validates :name, :presence => true, :uniqueness => true
+
+  serialize :properties, Hash
```

Comment:
The reason I ask is that the attributes should kind of look like columns, and columns are in a specific format...no capital letters, can't start with a number, etc.  method_missing won't get triggered properly in certain cases if they name their attributes wrong.